index.js

```diff
@@ -53,24 +52,73 @@ const sendRequest = (async (method, params) => {
     return result
   })
 })
-  
-const fetchBlock = async (block_index) => {
-  let blockHash = await sendRequest('getblockhash', [block_index])
+
+const fetchBlockHashByIndex = async (blockIndex) => {
+  return await sendRequest('getblockhash', [blockIndex])
+}
+
+const fetchBlockByHash = async (blockHash) => {
   return await sendRequest('getblock', [blockHash, 2])
 }
 
+const fetchBlockByIndex = async (blockIndex) => {
+  let blockHash = await fetchBlockHashByIndex(blockIndex)
+  return await fetchBlockByHash(blockHash)
+}
+
+async function rollbackOrphans() {
+  let blocks = []
+  let lastProcessedHash = lastProcessed.blockHash
+  let lastExportedBlock = await fetchBlockByHash(lastProcessedHash);
+
+  while (lastExportedBlock.confirmations === -1) {
+    blocks.unshift(lastExportedBlock)
+    lastProcessedHash = lastExportedBlock.previousblockhash
+    lastExportedBlock = await fetchBlockByHash(lastProcessedHash);
+  }
+  if(blocks.length) {
+
+    // TODO find if we need to decrement this counter
+    // TODO in that case it would be better to change to Gauge
+
+    // const blocks = await Promise.all(blocks).map(async (block) => {
+    //   metrics.downloadedTransactionsCounter.dec(block.tx.length)
+    //   metrics.downloadedBlocksCounter.dec()
+
+    //   return block
+    // })
```

Comment:
I don't think there is a need to touch the metrics here. May be in the future we can add a new metric, which tracks the number of soft forks that we detected and handled.